dali/python/nvidia/dali/pipeline.py

```diff
@@ -293,16 +363,28 @@ def _prepare_graph(self):
 
         # Add the ops to the graph and build the backend
         related_logical_id = {}
+        self._ops = []
         while ops:
             op = ops.pop()
+            self._ops.append(op)
             if op.relation_id not in related_logical_id:
                 related_logical_id[op.relation_id] = self._pipe.AddOperator(op.spec, op.name)
             else:
                 self._pipe.AddOperator(op.spec, op.name, related_logical_id[op.relation_id])
         self._prepared = True
+        self._setup_input_callbacks()
         self._names_and_devices = [(e.name, e.device) for e in outputs]
 
-    def build(self):
+    def _setup_input_callbacks(self):
+        from nvidia.dali.external_source import _is_external_source_with_callback
+        groups = set()
+        for op in self._ops:
+            if _is_external_source_with_callback(op):
+                group = op._group
+                groups.add(group)
+        self._input_callbacks = list(groups)
+
+    def build(self, define_graph = None):
```

Comment:
This adds yet another way to build a pipeline and at some point we will start causing confusion without clear doc how to do something.

Especially as this argument will now appear in docs, but just not documented.